renderdoc/driver/d3d12/d3d12_replay.cpp

```diff
@@ -900,34 +865,49 @@ void D3D12Replay::FillResourceView(D3D12Pipe::View &view, const D3D12Descriptor
   }
 }
 
-void D3D12Replay::FillRegisterSpaces(const D3D12RenderState::RootSignature &rootSig,
-                                     const ShaderBindpointMapping &mapping,
-                                     rdcarray<D3D12Pipe::RegisterSpace> &dstSpaces,
-                                     D3D12_SHADER_VISIBILITY visibility)
+static_assert((int32_t)RootElementVisibility::All == (int32_t)D3D12_SHADER_VISIBILITY_ALL,
+              "RootElementVisibility mismatch");
+static_assert((int32_t)RootElementVisibility::Vertex == (int32_t)D3D12_SHADER_VISIBILITY_VERTEX,
+              "RootElementVisibility mismatch");
+static_assert((int32_t)RootElementVisibility::Hull == (int32_t)D3D12_SHADER_VISIBILITY_HULL,
+              "RootElementVisibility mismatch");
+static_assert((int32_t)RootElementVisibility::Domain == (int32_t)D3D12_SHADER_VISIBILITY_DOMAIN,
+              "RootElementVisibility mismatch");
+static_assert((int32_t)RootElementVisibility::Geometry == (int32_t)D3D12_SHADER_VISIBILITY_GEOMETRY,
+              "RootElementVisibility mismatch");
+static_assert((int32_t)RootElementVisibility::Pixel == (int32_t)D3D12_SHADER_VISIBILITY_PIXEL,
```

Comment:
For not any particularly good reasons I define `RDCCOMPILE_ASSERT` for consistency with `RDCASSERT`, which should be preferred.